test/inductor/test_pattern_matcher.py

```diff
@@ -178,6 +178,66 @@ def fn2(a, b, c):
             self._test_fused_int_mm_mul_impl(fn1, args, True)
             self._test_fused_int_mm_mul_impl(fn2, args, True)
 
+    def test_duplicate_search(self):
+        from typing import Callable, Iterable
+
+        import torch
+        from torch._inductor.pattern_matcher import (
+            fwd_only,
+            PatternMatcherPass,
+            register_replacement,
+        )
+
+        def pattern1(x: torch.Tensor) -> torch.Tensor:
+            return x + 1
+
+        def replacement1(x: torch.Tensor) -> torch.Tensor:
+            return x - 1
+
+        def pattern2(x: torch.Tensor) -> torch.Tensor:
+            return x + 2
+
+        def replacement2(x: torch.Tensor) -> torch.Tensor:
+            return x - 2
+
+        patterns = PatternMatcherPass()
+        torch.set_default_device("cuda")
```

Comment:
Wouldn't that potentially affect other tests in a very weird ways? Shouldn't one better use `with torch.device("cuda"):` or just alloc next tensor on the device you want?